### PR TITLE
Androapp 5724 remove organization unit in tei event ds card if user only has access to one org unit

### PR DESCRIPTION
**Pull request by siddh1004**
## Description
IF the user has access to more than one org unit in capture OR search tree  - DISPLAY ORG UNIT IN CARD

IF the user has access to only ONE org unit in capture AND search tree  - DON’T DISPLAY ORG UNIT (Enrolled in/Registered in)

[ jira issue ](https://dhis2.atlassian.net/browse/ANDROAPP-5724)

## Solution description
If this PR is a fix include a brief description on how the issue is solved.
## Covered unit test cases
Describe the tests that you ran to verify your changes.
## Where did you test this issue?
- [x] Smartphone Emulator
- [ ] Tablet Emulator
- [ ] Smartphone
- [ ] Tablet
## Which Android versions did you test this issue?
- [ ] 4.4
- [ ] 5.X - 6.X
- [ ] 7.X
- [ ] 8.X
- [ ] 9.X - 10.X
- [x] 11.X - 13.X
- [ ] Other
## Checklist
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code

### CODE REVIEW

File: app/src/main/java/org/dhis2/usescases/programEventDetail/ProgramEventMapper.kt [hunk 1]
```diff
@@ -32,6 +32,10 @@ class ProgramEventMapper @Inject constructor(
         val programStage =
             d2.programModule().programStages().uid(event.programStage()).blockingGet()
 
+        val dataSetOrgUnitNumber = d2.organisationUnitModule().organisationUnits()
+            .byProgramUids(listOf(event.program()!!))
+            .blockingGet().size
+
         crashReportController.addBreadCrumb(
```
Comment (andresmr) [inline app/src/main/java/org/dhis2/usescases/programEventDetail/ProgramEventMapper.kt:39]: please, can you remove this breadcrumb, we no longer need it.

File: app/src/main/java/org/dhis2/usescases/programEventDetail/ProgramEventMapper.kt [hunk 2]
```diff
@@ -62,6 +66,7 @@ class ProgramEventMapper @Inject constructor(
             },
             nameCategoryOptionCombo =
             getCategoryComboFromOptionCombo(event.attributeOptionCombo())?.displayName(),
+            displayOrgUnitName = dataSetOrgUnitNumber > 1,
```
Comment (andresmr) [inline app/src/main/java/org/dhis2/usescases/programEventDetail/ProgramEventMapper.kt:69]: EventViewModel is a Domain model, indeed it is in `package org.dhis2.commons.data`, `displayOrgUnitName` is more related to presentation logic, so I will evaluate it in a UseCase as you are going to reuse it, and the inject the use case in the classes where you generate the UI model like `ListCardUiModel`. 
You can follow the [Android guidelines](https://developer.android.com/topic/architecture/domain-layer) to generate use cases.

**Review by andresmr — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/dhis2/usescases/programEventDetail/ProgramEventMapper.kt (hunk 1 above)
Comment (siddh1004) [reply app/src/main/java/org/dhis2/usescases/programEventDetail/ProgramEventMapper.kt]: breadcrumb removed

File: app/src/main/java/org/dhis2/usescases/programEventDetail/ProgramEventMapper.kt (hunk 2 above)
Comment (siddh1004) [reply app/src/main/java/org/dhis2/usescases/programEventDetail/ProgramEventMapper.kt]: `displayOrgUnitName` moved to UI model

**Review by andresmr — APPROVED** [review]

**Review by mmmateos — APPROVED** [review]

**Pull request merged**